libsignal-service/src/push_service.rs

```diff
@@ -354,7 +354,7 @@ pub enum ServiceError {
 
 #[cfg_attr(feature = "require-send", async_trait::async_trait)]
 #[cfg_attr(not(feature = "require-send"), async_trait::async_trait(?Send))]
-pub trait PushService {
+pub trait PushService: MaybeSend {
```

Comment:
Hmm, this still has the awkward `MaybeSend`, but I suppose that without the trait-aliases it's going to be a lot of code copy-paste to get that out.

Reply:
Indeed, I don't think there is any other way to conditionally require `Send` while staying on stable. It's either the weird trait, or you have to copy paste the entire `pub trait PushService { /* ... */ }`.

Reply:
You could probably macro it out, if you're up to that, but I'd wait with that until @gferon gives some fb :)

Reply:
Let's not macro it. We might want a tracking issue for when trait aliases stabilize and are stable for ~6 months.